### ISSUE TITLE
Not possible to navigate to IPv6 Link-Local.

### ISSUE THREAD

**Issue by cetex** [qutebrowser/qutebrowser#4668]
**Version info (see `:version`)**: v1.5.1

**Does the bug happen if you start with `--temp-basedir`?**: yes

**Description**
Can't navigate to ipv6 link-local addresses like: http://[fe80::a236:9fff:fe35:5d4c%wlp3s0]/
I get redirected to duckduckgo instead.

Realized this today when trying to reach a firewall on lan.

**How to reproduce**
Try to open a url like: http://[fe80::a236:9fff:fe35:5d4c%wlp3s0]/
where fe80::a236:9fff:fe35:5d4c is the ipv6 address and wlp3s0 is the interface name.

RFC6874 at https://tools.ietf.org/html/rfc6874 describes different ways to enter an ipv6 address where the [address%interface] is the proposed solution.

**Comment by toofar** [comment]
```python
> from PyQt5.QtCore import QUrl
> u=QUrl("http://[fe80::a236:9fff:fe35:5d4c%wlp3s0]/")
> u
PyQt5.QtCore.QUrl('')
> u.isValid()
False
```
Here is the relevant upstream bug https://bugreports.qt.io/browse/QTBUG-46298 you can vote for that if you have an account. It hasn't seen any movement for a few years though.
I'm not sure if webengine could handle those URLs itself (if chromium can then it probably can) but there are far too many `QUrl` wrappers around that for us to be able to support it if Qt doesn't.

**Comment by The-Compiler** [comment]
Note that at least with QtWebEngine the QNetworkAccessManager part isn't relevant, only how QUrl handles this case. For that, there's already [QTBUG-25550](https://bugreports.qt.io/browse/QTBUG-25550) which got fixed in Qt 5.12 - and indeed, I can ccess that URL fine and it works with Qt 5.12:

```python
>>> from PyQt5.QtCore import QUrl
>>> u=QUrl("http://[fe80::a236:9fff:fe35:5d4c%wlp3s0]/")
>>> u
PyQt5.QtCore.QUrl('http://[fe80::a236:9fff:fe35:5d4c%25wlp3s0]/')
>>> u.isValid()
True
```

**Comment by toofar** [comment]
Whoops, obviously I was testing on an older Qt version. I don't have IPv6 enabled at the moment so can't look into this any further untill I reboot.

**Comment by The-Compiler** [comment]
I'm pretty sure it works with Qt 5.12 - but if you want to test it, @toofar, feel free :wink:

**Comment by toofar** [comment]
Well I did test it it and it doesn't appear to load the page. I tested in an arch chroot with Qt 5.12.2 (compiled 5.12.1 whatever that means), PyQt 5.12, qute 1.6.1.
> 22:05:52 DEBUG    url        urlutils:is_url:266 Checking if 'http://[fe80::62a4:4cff:feb2:b5a4%br0]/' is a URL (autosearch=naive).
> 22:05:52 DEBUG    url        urlutils:is_url:307 url = True
> 22:05:52 DEBUG    webview    tabbedbrowser:tabopen:464 Creating new tab with URL PyQt5.QtCore.QUrl('http://[fe80::62a4:4cff:feb2:b5a4%25br0]/'), background False, related False, idx None
> 22:05:52 DEBUG    webview    browsertab:_on_before_load_started:955 Going to start loading: http://[fe80::62a4:4cff:feb2:b5a4%25br0]/
> 22:05:52 DEBUG    webview    tabbedbrowser:on_title_changed:623 Changing title for idx 2 to 'http://[fe80::62a4:4cff:feb2:b5a4%25br0]/'
> 22:05:52 DEBUG    webview    browsertab:_on_before_load_started:955 Going to start loading: http://[fe80::62a4:4cff:feb2:b5a4%25br0]/
> 22:05:52 DEBUG    webview    tabbedbrowser:on_title_changed:623 Changing title for idx 2 to 'http://[fe80::62a4:4cff:feb2:b5a4%25br0]/'

Then there is some tab change signals but nothing else. There is no URL or progress bar in the status bar, just the scroll and index widgets. It looks I am not getting any cur_url_changed or load status signals, or navigation request. That address works with ping and the URL works with curl (actually it works with curl 7.58.0 in debian but not with 7.64.0 in arch...). The DHCP assigned "global" scoped address loads happily in the browser (on Qt5.11 too).

I'm pretty sure it isn't a qutebrowser issue but I'm not sure if it is my setup or webengine. With this I get a blank white screen
```cpp
#include <iostream>
#include <QApplication>
#include <QWebEngineView>
#include <QUrl>

int main(int argc, char *argv[])
{
    QApplication app(argc, argv);
    QWebEngineView view;

    std::string url = "http://[fe80::62a4:4cff:feb2:b5a4%br0]/";
    if (argc > 1) {
      url = argv[1];
    }
    std::cout << QUrl(url.c_str()).toDisplayString().toStdString() << std::endl;
    view.load(QUrl(url.c_str()));
    view.show();
    return app.exec();
}
```

@cetex can you test with qutebrowser with Qt5.12? You can install a standalone Qt and other dependancies if your distro doesn't have it via tox: https://qutebrowser.org/doc/install.html#_installing_dependencies_including_qt

**Comment by The-Compiler** [comment]
@toofar Out of curiosity, does it work in Chromium for you? By the way, if you want a minimal C++ QtWebEngine example, there's one in qutebrowser's repo as well: https://github.com/qutebrowser/qutebrowser/tree/master/scripts/testbrowser/cpp/webengine

**Comment by toofar** [comment]
No, chromium does the same thing as google-chrome; interprets the URL as a search string.
